src/components/connections/SummaryDetails.tsx

```diff
@@ -91,6 +93,157 @@ const SummaryDetails = ({
     },
   ];
 
+  const generateCSV = () => {
```

Comment:
suggestion (non-blocking): This could potentially be moved to a service and made generic so that we can use it both here and in the summary header